packages/core/src/locales/fi.ts

```diff
@@ -0,0 +1,197 @@
+import type { $ZodStringFormats } from "../checks.js";
+import type * as errors from "../errors.js";
+import type { $ZodTypeDef } from "../schemas.js";
+import * as util from "../util.js";
+
+const Sizable: Record<string, { unit: string; subject: string }> = {
+  string: { unit: "merkkiä", subject: "merkkijonon" },
+  file: { unit: "tavua", subject: "tiedoston" },
+  array: { unit: "alkiota", subject: "listan" },
+  set: { unit: "alkiota", subject: "joukon" },
+  number: { unit: "", subject: "luvun" },
+  bigint: { unit: "", subject: "suuren kokonaisluvun" },
+  int: { unit: "", subject: "kokonaisluvun" },
+  date: { unit: "", subject: "päivämäärän" },
+};
+
+function getSizing(origin: string): { unit: string; subject: string } | null {
+  return Sizable[origin] ?? null;
+}
+
+const TypeNames: { [k in $ZodTypeDef["type"] | (string & {})]?: string } = {
+  string: "merkkijono",
+  number: "luku",
+  boolean: "totuusarvo",
+  bigint: "suuri kokonaisluku",
+  symbol: "symboli",
+  null: "tyhjä",
+  undefined: "määrittelemätön",
+  date: "päivämäärä",
+  object: "objekti",
```

Comment:
Kind of hate this term, especially if these are displayed to non-technical users.
But I am not entirely sure where to draw the line. Similar to `map` comment below, I think in this case `"tietue"` _might also work_...